userspace/libsinsp/test/public_sinsp_API/ppm_sc_codes.cpp

```diff
@@ -0,0 +1,947 @@
+	ASSERT_EQ(all_events.size(), PPM_EVENT_MAX);
+	for(int i = 0; i < PPM_EVENT_MAX; i++)
+	{
+		ASSERT_TRUE(all_events.contains((ppm_event_code)i)) << "\n- The event '" << scap_get_event_info_table()[i].name << "' is not present inside the all event set" << std::endl;
+	}
+}
+
+TEST(ppm_sc_API, all_event_names)
+{
+	/* Here we want all events' names also the ones associated with generic events, so the syscalls
+	 * names, but we don't want the "syscall" event name associated with `GENERIC_E`/`GENERIC_X` events extracted from the event table.
+	 */
+	auto events_names = test_utils::unordered_set_to_ordered(libsinsp::events::event_set_to_names(libsinsp::events::all_event_set()));
+	/* `NA*` events were now removed so we don't want them again, all other syscall names have no events associated so they shouldn't be in this set */
+	std::set<std::string> some_not_desired_names{"syscall", "pipe2", "inotify_init1", "eventfd2", "signalfd4", "ugetrlimit", "fcntl64", "sendfile64", "setresuid32", "setresgid32", "setuid32", "setgid32", "getuid32", "geteuid32", "getgid32", "getegid32", "getresuid32", "getresgid32", "NA1", "NA2", "NA3", "NA4", "NA5", "NA6"};
+	ASSERT_NOT_CONTAINS(events_names, some_not_desired_names);
+
+	/* We count old version events to be sure about the final number of names we should expect */
+	int old_versions_events = 0;
+	std::set<std::string> all_expected_events_names = {};
+
+	/* We skip `syscall` name associated with `GENERIC_E`/`GENERIC_X` */
+	for(int evt = 2; evt < PPM_EVENT_MAX; evt++)
+	{
+		if(libsinsp::events::is_old_version_event((ppm_event_code)evt))
+		{
+			old_versions_events++;
+		}
+		all_expected_events_names.insert(scap_get_event_info_table()[evt].name);
+	}
+
+	libsinsp::events::set<ppm_event_code> generic_events{PPME_GENERIC_E, PPME_GENERIC_X};
+	std::vector<uint8_t> generic_syscalls(PPM_SC_MAX, 0);
+	ASSERT_EQ(scap_get_ppm_sc_from_events(generic_events.data(), generic_syscalls.data()), SCAP_SUCCESS);
+	for(int ppm_sc = 0; ppm_sc < PPM_SC_MAX; ppm_sc++)
+	{
+		if(generic_syscalls[ppm_sc])
+		{
+			all_expected_events_names.insert(scap_get_syscall_info_table()[ppm_sc].name);
+		}
+	}
+	ASSERT_NAMES_EQ(events_names, all_expected_events_names);
+	/* To obtain the right size of the event names we need to divide by 2 the total number of events.
+	 * Events are almost all paired, and when they are not paired dividing by 2 we remove the `NA` entries.
+	 * Since we consider the `NA` a valid name we need to add it to the set, so `+1`
+	 * We don't want the name "syscall" associated with  `PPME_GENERIC_E` and `PPME_GENERIC_E`, so `-1`. `-1` and not `-2` because we have already divided by 2.
+	 * We need to remove all the old version events because their names are just a replica of current events ones. `/2` because we have already divided by 2.
+	 * Finally we need to add the GENERIC names.
+	 */
+	ASSERT_EQ(events_names.size(), (PPM_EVENT_MAX / 2) + 1 - 1 - old_versions_events / 2 + GENERIC_SYSCALLS_NUM);
```

Comment:
I see why you typed out + 1 -1 , but we can remove it ... mind adding more parentheses to make it clearer what the intended math is?
```suggestion
	ASSERT_EQ(events_names.size(), (PPM_EVENT_MAX / 2) - (old_versions_events / 2) + GENERIC_SYSCALLS_NUM);
```